### PR TITLE
Add c# wrapper

### LINKED ISSUE #50: Create some sort of C# adapter (wrapper) to hide tons of dynamic casting and make working with the console from C# easier.
More info: https://github.com/quentincaffeino/godot-console/issues/48#issuecomment-769022507

### PR DESCRIPTION
**Pull request by ryan-linehan**
I wanted a wrapper for my own project. So figured I'd create a PR here for it as well which should close #50 . Let me know if I need to change anything or if I missed anything. Mostly went for basic functionality that I needed in my own project but I am happy to expand if necessary. Ended up adding my own demo project for csharp which I can remove if wanted. The first commit is the demo project. The second commit updates the README.md and adds the CSharpConsole.cs file to a folder instead of `godot-console\addons\quentincaffeino\console\src\CSharp` Hope this can help!

### CODE REVIEW

File: addons/quentincaffeino/console-csharp/src/Console.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+{    
+	CanvasLayer _console;
+	public override void _Ready()
+	{
+		_console = GetTree().Root.GetNode<CanvasLayer>("Console");
+	}	
+
+	public ConsoleCommand AddCommand(string name, Godot.Object target, string targetMethodName) {
+		Godot.Object consoleCommand = _console.Call("add_command", name, target, targetMethodName) as Godot.Object;
+		return new ConsoleCommand(consoleCommand);
```
[the remark this answers is not in the document]
Comment (ryan-linehan) [reply addons/quentincaffeino/console-csharp/src/Console.cs]: Changed

File: addons/quentincaffeino/console/DefaultActions.gd [hunk 2]
```diff
@@ -5,10 +5,10 @@ const CONSOLE_TOGGLE = "quentincaffeino_console_toggle"
 const console_toggle_props = {
   "name": CONSOLE_TOGGLE,
   "events": [
-    {
-      "command": true,
-      "scancode": KEY_QUOTELEFT,
-    }
+	{
```
[the remark this answers is not in the document]
Comment (ryan-linehan) [reply addons/quentincaffeino/console/DefaultActions.gd]: Fixed!

**Comment by ryan-linehan** [comment]
Did you get a chance to look at the edits I made?

**Review by quentincaffeino — APPROVED** [review]

**Comment by quentincaffeino** [comment]
@ryan-linehan, yes, however I haven't been able to test it yet. I just run demo-csharp project and when try to run the game I get:

```
ERROR: Script does not inherit a Node: res://addons/quentincaffeino/console-csharp/src/Console.cs.
   at: _create_autoload (editor/editor_autoload_settings.cpp:356)
ERROR: Script does not inherit a Node: res://addons/quentincaffeino/console-csharp/src/Console.cs.
   at: _create_autoload (editor/editor_autoload_settings.cpp:356)
ERROR: Condition "!info->node" is true. Continuing.
   at: update_autoload (editor/editor_autoload_settings.cpp:488)
ERROR: Condition "!named_globals.has(p_name)" is true.
   at: remove_named_global_constant (modules/gdscript/gdscript.cpp:1371)
ERROR: Script does not inherit a Node: res://addons/quentincaffeino/console-csharp/src/Console.cs.
   at: _create_autoload (editor/editor_autoload_settings.cpp:356)
ERROR: Condition "!info->node" is true. Continuing.
   at: update_autoload (editor/editor_autoload_settings.cpp:488)
System.IO.FileNotFoundException: Cannot find binary for 'MSBuild (Mono)'
  at GodotTools.Build.MsBuildFinder.FindMsBuild () [0x00206] in <6e8d1f7535854a81992111020a024e35>:0 
  at GodotTools.Build.BuildSystem.LaunchBuild (GodotTools.Build.BuildInfo buildInfo, System.Action`1[T] stdOutHandler, System.Action`1[T] stdErrHandler) [0x00014] in <6e8d1f7535854a81992111020a024e35>:0 
  at GodotTools.Build.BuildSystem.Build (GodotTools.Build.BuildInfo buildInfo, System.Action`1[T] stdOutHandler, System.Action`1[T] stdErrHandler) [0x00000] in <6e8d1f7535854a81992111020a024e35>:0 
  at GodotTools.Build.BuildManager.Build (GodotTools.Build.BuildInfo buildInfo) [0x00066] in <6e8d1f7535854a81992111020a024e35>:0 
ERROR: An EditorPlugin build callback failed.
   at: call_build (editor/editor_node.cpp:5424)
```

I'm on Godot 3.4 stable-mono

**Comment by quentincaffeino** [comment]
I've also been thinking about generating methods to omit having to support two code-bases but that turned out to be some where between unusable and impossible to do good.

**Comment by quentincaffeino** [comment]
Should've read error completely first, seems that it is missing the compiler, I'll install it and get back to you in a moment.

**Comment by quentincaffeino** [comment]
I was unsuccessful in solving this problem, will need to wait for a fix or till there is a solution https://github.com/godotengine/godot/issues/55610

**Comment by quentincaffeino** [comment]
@all-contributors please add @ryan-linehan for code

**Pull request merged**